# 1. Data Extraction

In this notebook we will extract our data and analyse it. For that purpose, we are importing from our library where we define the
```bcrp_dataframe``` dataframe. This function will allows us to use the API interface of the Central Bank of Reserve of Peru (BCRP) to automatically create a pandas dataframe with the necessary codes.

## 1.1 Libraries

We import the necessary libraries, including our own library in the modules file

In [1]:
# Warnings
import warnings
warnings.filterwarnings("ignore")

# Basic Libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import requests
import seaborn as sns
from scipy import stats
from functools import reduce

# Statsmodels
import statsmodels.api as sm
import pmdarima as pmd
from pmdarima.arima import auto_arima
from statsmodels.tsa.api import VAR
from statsmodels.tsa.vector_ar.var_model import VARResults
from statsmodels.tsa.statespace.sarimax import SARIMAX
from statsmodels.tsa.arima.model import ARIMA
from statsmodels.tsa.stattools import adfuller
from statsmodels.tsa.seasonal import STL

# Machine Learning models
from sklearn.ensemble import RandomForestRegressor, RandomForestClassifier
from sklearn.model_selection import train_test_split, GridSearchCV, TimeSeriesSplit
from sklearn.linear_model import Ridge, Lasso, ElasticNet, ElasticNetCV, LinearRegression
from sklearn.linear_model import LinearRegression
from sklearn.metrics import (
    mean_absolute_error,
    mean_squared_error,
    mean_absolute_percentage_error,
    median_absolute_error,
    r2_score,
    precision_score

)

from xgboost import XGBRegressor



In [2]:
# We import our own functions
import sys
sys.path.append('../../..')  # Move two levels up to the project root
from modules.functions import *

## 1.2 Extraction
We define our inputs and apply them the ```bcrp_dataframe``` function in order to obtain the pandas dataframe with the corresponding series

We define the following inputs:

    series     = the code of the series we are going to extract
    start_date = the starting date, when the BCRP starts using the interest rate as a policy measure
    end_date   = December 2019
    freq       = Monthly frequency

### df_1
We can now create the first dataframe with the ```bcrp_dataframe``` function. This dataframe contains all Consumer Price Index variables in monthly % change. This variables are then seasonally adjusted using the 

In [3]:
series     = ['PN01271PM', 'PN01280PM', 'PN01282PM', 'PN01278PM', 'PN09817PM','PN09816PM', 'PN01276PM', 'PN01313PM', 'PN01314PM',  
             'PN01315PM', 'PN09818PM','PN01286PM']
start_date = '2003-01'
end_date   = '2023-12'
freq       = 'Mensual'

In [4]:
df_1 = bcrp_dataframe( series , start_date , end_date , freq )
df_1.head()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
Fecha                                                                   
2003-01-01                                           0.231420           
2003-02-01                                           0.468825           
2003-03-01                                           1.117780           
2003-04-01                                          -0.050857           
2003-05-01                                          -0.032025           

            Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
Fecha                                                                              
2003-01-01                                           0.179002                      
2003-02-01                                           0.135374                      
2003-03-01                                           0.400108                      
2003-04-01                                           0.112646                      
2003-05-01                                          -0.346022                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
Fecha                                                                                 
2003-01-01                                           0.270381                         
2003-02-01                                           0.708770                         
2003-03-01                                           1.623551                         
2003-04-01                                          -0.182128                         
2003-05-01                                           0.181194                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
Fecha                                                                              
2003-01-01                                          -0.004192                      
2003-02-01                                           0.128185                      
2003-03-01                                           0.157515                      
2003-04-01                                           0.074363                      
2003-05-01                                           0.011232                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
Fecha                                                                                 
2003-01-01                                           0.593592                         
2003-02-01                                           0.992255                         
2003-03-01                                           2.564665                         
2003-04-01                                          -0.260881                         
2003-05-01                                          -0.102900                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
Fecha                                                                                       
2003-01-01                                           0.506054                               
2003-02-01                                           0.722192                               
2003-03-01                                           0.357386                               
2003-04-01                                          -0.265865                               
2003-05-01                                          -0.139973                               

            Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía  \
Fecha                                                                                           
2003-01-01                                          -0.111733                                   
2003-02-01                                           0.228145                                   
2003-03-01                                           1.938313 

In [5]:
df_1 = get_trend(df_1)

df_1.head()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
Fecha                                                                   
2003-01-01                                           0.092926           
2003-02-01                                           0.112976           
2003-03-01                                           0.133060           
2003-04-01                                           0.153179           
2003-05-01                                           0.173188           

            Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
Fecha                                                                              
2003-01-01                                          -0.090958                      
2003-02-01                                          -0.053866                      
2003-03-01                                          -0.016576                      
2003-04-01                                           0.020809                      
2003-05-01                                           0.058179                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
Fecha                                                                                 
2003-01-01                                           0.221921                         
2003-02-01                                           0.230003                         
2003-03-01                                           0.238007                         
2003-04-01                                           0.246004                         
2003-05-01                                           0.253822                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
Fecha                                                                              
2003-01-01                                           0.023684                      
2003-02-01                                           0.029531                      
2003-03-01                                           0.035395                      
2003-04-01                                           0.041241                      
2003-05-01                                           0.047045                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
Fecha                                                                                 
2003-01-01                                           0.208767                         
2003-02-01                                           0.247979                         
2003-03-01                                           0.287271                         
2003-04-01                                           0.326699                         
2003-05-01                                           0.365928                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
Fecha                                                                                       
2003-01-01                                          -0.178262                               
2003-02-01                                          -0.113158                               
2003-03-01                                          -0.048449                               
2003-04-01                                           0.015669                               
2003-05-01                                           0.078897                               

            Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía  \
Fecha                                                                                           
2003-01-01                                           0.421416                                   
2003-02-01                                           0.387611                                   
2003-03-01                                           0.354348 

### df_2
We create the second dataframe with the ```bcrp_dataframe``` function. This dataframe contains rate variables. We use this variables in levels. It is not necessary to differentiate them.

In [6]:
series     = ['PN00493MM', 'PD04722MM']
start_date = '2003-09'
end_date   = '2023-12'
freq       = 'Mensual'

In [7]:
df_2 = bcrp_dataframe( series , start_date , end_date , freq )
df_2.head()

Tasas de interés del Banco Central de Reserva  - Tasa de Encaje  \
Fecha                                                                         
2003-09-01                                          11.014920                 
2003-10-01                                          10.349944                 
2003-11-01                                          11.690608                 
2003-12-01                                          11.391178                 
2004-01-01                                          10.634030                 

            Tasas de interés del Banco Central de Reserva  - Tasa de Referencia de la Política Monetaria  
Fecha                                                                                                     
2003-09-01                                               2.75                                             
2003-10-01                                               2.75                                             
2003-11-01                                               2.50                                             
2003-12-01                                               2.50                                             
2004-01-01                                               2.50

### df_3
We create the third dataframe with the ```bcrp_dataframe``` function. This dataframe contains monetary variables. We differentiate those variables in order to get their monthly % change

In [8]:
series     = ['PN00495MM', 'PN06481IM', 'PN02125PM']
start_date = '2003-09'
end_date   = '2023-12'
freq       = 'Mensual'

In [9]:
df_3 = bcrp_dataframe( series , start_date , end_date , freq )
df_3 = df_3.pct_change()
df_3 = df_3.dropna()
df_3.head()

Emisión primaria y multiplicador (millones S/) - Circulante Desestacionalizado  \
Fecha                                                                                        
2003-10-01                                           0.010225                                
2003-11-01                                           0.011445                                
2003-12-01                                           0.016607                                
2004-01-01                                           0.019490                                
2004-02-01                                           0.017003                                

            Liquidez internacional del BCRP - RIN - Reservas Internacionales Netas (millones US$)  \
Fecha                                                                                               
2003-10-01                                           0.005542                                       
2003-11-01                                           0.050288                                       
2003-12-01                                          -0.010532                                       
2004-01-01                                           0.036233                                       
2004-02-01                                          -0.005743                                       

            Remuneraciones - Remuneración Mínima Vital - Índice Real (base 1994 = 100)  
Fecha                                                                                   
2003-10-01                                           0.056953                           
2003-11-01                                          -0.001674                           
2003-12-01                                          -0.005608                           
2004-01-01                                          -0.005346                           
2004-02-01                                          -0.010744

In [10]:
df = df_1.join(df_2).join(df_3)
df.dropna(inplace=True)
df.head()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
Fecha                                                                   
2003-10-01                                           0.266135           
2003-11-01                                           0.283860           
2003-12-01                                           0.302840           
2004-01-01                                           0.315529           
2004-02-01                                           0.324218           

            Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
Fecha                                                                              
2003-10-01                                           0.241045                      
2003-11-01                                           0.277553                      
2003-12-01                                           0.315870                      
2004-01-01                                           0.343327                      
2004-02-01                                           0.365682                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
Fecha                                                                                 
2003-10-01                                           0.283686                         
2003-11-01                                           0.288313                         
2003-12-01                                           0.293891                         
2004-01-01                                           0.296279                         
2004-02-01                                           0.295321                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
Fecha                                                                              
2003-10-01                                           0.075596                      
2003-11-01                                           0.081709                      
2003-12-01                                           0.088471                      
2004-01-01                                           0.093899                      
2004-02-01                                           0.098356                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
Fecha                                                                                 
2003-10-01                                           0.545433                         
2003-11-01                                           0.578708                         
2003-12-01                                           0.614307                         
2004-01-01                                           0.636747                         
2004-02-01                                           0.650808                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
Fecha                                                                                       
2003-10-01                                           0.375007                               
2003-11-01                                           0.433341                               
2003-12-01                                           0.494624                               
2004-01-01                                           0.528569                               
2004-02-01                                           0.548055                               

            Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía  \
Fecha                                                                                           
2003-10-01                                           0.140042                                   
2003-11-01                                           0.109075                                   
2003-12-01                                           0.077290 

In [11]:
df.tail()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
Fecha                                                                   
2023-08-01                                           0.182324           
2023-09-01                                           0.133933           
2023-10-01                                           0.085365           
2023-11-01                                           0.036736           
2023-12-01                                          -0.011845           

            Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
Fecha                                                                              
2023-08-01                                           0.263821                      
2023-09-01                                           0.237390                      
2023-10-01                                           0.210974                      
2023-11-01                                           0.184472                      
2023-12-01                                           0.157822                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
Fecha                                                                                 
2023-08-01                                           0.147377                         
2023-09-01                                           0.089467                         
2023-10-01                                           0.031289                         
2023-11-01                                          -0.026935                         
2023-12-01                                          -0.085018                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
Fecha                                                                              
2023-08-01                                           0.308731                      
2023-09-01                                           0.281829                      
2023-10-01                                           0.254555                      
2023-11-01                                           0.226933                      
2023-12-01                                           0.198984                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
Fecha                                                                                 
2023-08-01                                          -0.071459                         
2023-09-01                                          -0.163217                         
2023-10-01                                          -0.254697                         
2023-11-01                                          -0.345597                         
2023-12-01                                          -0.435639                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
Fecha                                                                                       
2023-08-01                                           0.175741                               
2023-09-01                                           0.099223                               
2023-10-01                                           0.022078                               
2023-11-01                                          -0.055512                               
2023-12-01                                          -0.133358                               

            Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía  \
Fecha                                                                                           
2023-08-01                                           0.189515                                   
2023-09-01                                           0.165369                                   
2023-10-01                                           0.141472 

## 1.3 Data Inspection
We inspect the df. We first verify that all values are non-null. The, we apply the ```describe``` function to see the main variables.

In [12]:
df.isna().sum()

Índice de precios Lima Metropolitana (var% mensual) - IPC                                       0
Índice de precios Lima Metropolitana (var% mensual) - IPC Transables                            0
Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables                         0
Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente                            0
Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente                         0
Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía                   0
Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía               0
Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Bebidas                   0
Índice de precios Lima Metropolitana (var% mensual) - IPC sin Alimentos y Bebidas               0
Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente Sin Alimentos y Bebidas    0
Índice de precios Li

In [13]:
df.describe()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
count                                         243.000000           
mean                                            0.264854           
std                                             0.145368           
min                                            -0.011845           
25%                                             0.159071           
50%                                             0.247743           
75%                                             0.319375           
max                                             0.679208           

       Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
count                                         243.000000                      
mean                                            0.246279                      
std                                             0.150310                      
min                                            -0.015181                      
25%                                             0.142685                      
50%                                             0.211901                      
75%                                             0.296747                      
max                                             0.705393                      

       Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
count                                         243.000000                         
mean                                            0.276922                         
std                                             0.156400                         
min                                            -0.085018                         
25%                                             0.167191                         
50%                                             0.264602                         
75%                                             0.347933                         
max                                             0.711927                         

       Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
count                                         243.000000                      
mean                                            0.252162                      
std                                             0.111051                      
min                                             0.075596                      
25%                                             0.162488                      
50%                                             0.254535                      
75%                                             0.306935                      
max                                             0.532530                      

       Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
count                                         243.000000                         
mean                                            0.286713                         
std                                             0.272234                         
min                                            -0.435639                         
25%                                             0.142472                         
50%                                             0.229393                         
75%                                             0.384886                         
max                                             1.065628                         

       Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
count                                         243.000000                               
mean                                            0.336332                               
std                                             0.234510                               
min                                            -0.133358                               
25%                                             0.16

We have 195 observation ranging from ```2003-10-01``` to ```2019-12-01```. The mean of monthly % change of all CPI variables is around 0.2. The mean of the lacing rate and the interest rate is 10.7% and 3.67%, respectively. The three monetary variables have a small monthly % change, around 0.01 and 0.001 for Minimum Wage index. 

## 1.4 Data adjustment
We will rename the columns for easier identification of the variables. We will also create a new dataframe with the lags of the variables. 

In [14]:
# New column names
columns = {
    'Índice de precios Lima Metropolitana (var% mensual) - IPC': 'CPI',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Transables': 'CPI Tradable',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables': 'CPI Non-Tradable',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente': 'CPI Core',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente': 'CPI Non-Core',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía': 'CPI Food and Energy',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía': 'CPI Excluding Food and Energy',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Bebidas': 'CPI Food and Beverages',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC sin Alimentos y Bebidas': 'CPI Excluding Food and Beverages',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente Sin Alimentos y Bebidas': 'CPI Core Excluding Food and Beverages',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Importado': 'CPI Imported',
    'Índice de precios Lima Metropolitana (var% mensual) - Índice de Precios al por Mayor': 'Wholesale Price Index',
    'Tasas de interés del Banco Central de Reserva  - Tasa de Encaje': 'Reserve Requirement Rate',
    'Tasas de interés del Banco Central de Reserva  - Tasa de Referencia de la Política Monetaria': 'Monetary Policy Rate',
    'Emisión primaria y multiplicador (millones S/) - Circulante Desestacionalizado': 'Circulating Currency Seasonally Adjusted (mill S/)',
    'Liquidez internacional del BCRP - RIN - Reservas Internacionales Netas (millones US$)': 'Net International Reserves (mill $)',
    'Remuneraciones - Remuneración Mínima Vital - Índice Real (base 1994 = 100)': 'Real Minimum Wage (Index)'
}

# We rename the columns so they are easier to analyse
df.rename(columns=columns, inplace=True)

In [15]:
df_lags = df.copy()

for variable in df_lags.columns[1:]:
    df_lags[f'{variable}_lag_1'] = df_lags[variable].shift()
    df_lags[f'{variable}_lag_2'] = df_lags[variable].shift(2)

In [16]:
# We delete contemporary variables
df_lags.drop(columns = ['CPI Tradable', 'CPI Non-Tradable', 'CPI Core', 'CPI Non-Core', 'CPI Food and Energy', 'CPI Excluding Food and Energy',
       'CPI Food and Beverages', 'CPI Excluding Food and Beverages','CPI Core Excluding Food and Beverages', 'CPI Imported',
       'Wholesale Price Index', 'Reserve Requirement Rate','Monetary Policy Rate','Circulating Currency Seasonally Adjusted (mill S/)',
       'Net International Reserves (mill $)', 'Real Minimum Wage (Index)'], inplace = True)

df_lags = df_lags.dropna()

## 1.5 Save Results
We save it to the ```input``` folder, where we can use it to do the forecasting in the next notebook.

In [17]:
df.to_csv('../../../input/df_raw_h23.csv')

In [18]:
df_lags.to_csv('../../../input/df_lags_h23.csv')

In [19]:
df_lags.tail()

CPI  CPI Tradable_lag_1  CPI Tradable_lag_2  \
Fecha                                                          
2023-08-01  0.182324            0.290249            0.316561   
2023-09-01  0.133933            0.263821            0.290249   
2023-10-01  0.085365            0.237390            0.263821   
2023-11-01  0.036736            0.210974            0.237390   
2023-12-01 -0.011845            0.184472            0.210974   

            CPI Non-Tradable_lag_1  CPI Non-Tradable_lag_2  CPI Core_lag_1  \
Fecha                                                                        
2023-08-01                0.204813                0.261591        0.335223   
2023-09-01                0.147377                0.204813        0.308731   
2023-10-01                0.089467                0.147377        0.281829   
2023-11-01                0.031289                0.089467        0.254555   
2023-12-01               -0.026935                0.031289        0.226933   

            CPI Core_lag_2  CPI Non-Core_lag_1  CPI Non-Core_lag_2  \
Fecha                                                                
2023-08-01        0.361269            0.020204            0.111347   
2023-09-01        0.335223           -0.071459            0.020204   
2023-10-01        0.308731           -0.163217           -0.071459   
2023-11-01        0.281829           -0.254697           -0.163217   
2023-12-01        0.254555           -0.345597           -0.254697   

            CPI Food and Energy_lag_1  ...  Reserve Requirement Rate_lag_1  \
Fecha                                  ...                                   
2023-08-01                   0.251427  ...                        5.617809   
2023-09-01                   0.175741  ...                        5.137951   
2023-10-01                   0.099223  ...                        5.269584   
2023-11-01                   0.022078  ...                        5.277209   
2023-12-01                  -0.055512  ...                        5.625714   

            Reserve Requirement Rate_lag_2  Monetary Policy Rate_lag_1  \
Fecha                                                                    
2023-08-01                        5.985237                        7.75   
2023-09-01                        5.617809                        7.75   
2023-10-01                        5.137951                        7.50   
2023-11-01                        5.269584                        7.25   
2023-12-01                        5.277209                        7.00   

            Monetary Policy Rate_lag_2  \
Fecha                                    
2023-08-01                        7.75   
2023-09-01                        7.75   
2023-10-01                        7.75   
2023-11-01                        7.50   
2023-12-01                        7.25   

            Circulating Currency Seasonally Adjusted (mill S/)_lag_1  \
Fecha                                                                  
2023-08-01                                          -0.017678          
2023-09-01                                          -0.002584          
2023-10-01                                           0.002271          
2023-11-01                                           0.006684          
2023-12-01                                          -0.002776          

            Circulating Currency Seasonally Adjusted (mill S/)_lag_2  \
Fecha                                                                  
2023-08-01                                          -0.011038          
2023-09-01                                          -0.017678          
2023-10-01                                          -0.002584          
2023-11-01                                           0.002271          
2023-12-01                                           0.006684          

            Net International Reserves (mill $)_lag_1  \
Fecha                                                   
2023-08-01                                   

In [20]:
df.tail()

CPI  CPI Tradable  CPI Non-Tradable  CPI Core  CPI Non-Core  \
Fecha                                                                          
2023-08-01  0.182324      0.263821          0.147377  0.308731     -0.071459   
2023-09-01  0.133933      0.237390          0.089467  0.281829     -0.163217   
2023-10-01  0.085365      0.210974          0.031289  0.254555     -0.254697   
2023-11-01  0.036736      0.184472         -0.026935  0.226933     -0.345597   
2023-12-01 -0.011845      0.157822         -0.085018  0.198984     -0.435639   

            CPI Food and Energy  CPI Excluding Food and Energy  \
Fecha                                                            
2023-08-01             0.175741                       0.189515   
2023-09-01             0.099223                       0.165369   
2023-10-01             0.022078                       0.141472   
2023-11-01            -0.055512                       0.117886   
2023-12-01            -0.133358                       0.094639   

            CPI Food and Beverages  CPI Excluding Food and Beverages  \
Fecha                                                                  
2023-08-01                0.263738                          0.125240   
2023-09-01                0.179682                          0.102272   
2023-10-01                0.094521                          0.079821   
2023-11-01                0.008508                          0.057913   
2023-12-01               -0.078087                          0.036539   

            CPI Core Excluding Food and Beverages  CPI Imported  \
Fecha                                                             
2023-08-01                               0.215058     -0.170008   
2023-09-01                               0.195769     -0.206148   
2023-10-01                               0.176261     -0.240150   
2023-11-01                               0.156596     -0.272242   
2023-12-01                               0.136819     -0.302719   

            Wholesale Price Index  Reserve Requirement Rate  \
Fecha                                                         
2023-08-01              -0.104763                  5.137951   
2023-09-01              -0.125837                  5.269584   
2023-10-01              -0.144366                  5.277209   
2023-11-01              -0.160344                  5.625714   
2023-12-01              -0.173843                  5.857378   

            Monetary Policy Rate  \
Fecha                              
2023-08-01                  7.75   
2023-09-01                  7.50   
2023-10-01                  7.25   
2023-11-01                  7.00   
2023-12-01                  6.75   

            Circulating Currency Seasonally Adjusted (mill S/)  \
Fecha                                                            
2023-08-01                                          -0.002584    
2023-09-01                                           0.002271    
2023-10-01                                           0.006684    
2023-11-01                                          -0.002776    
2023-12-01                                          -0.013468    

            Net International Reserves (mill $)  Real Minimum Wage (Index)  
Fecha                                                                       
2023-08-01                            -0.019404                  -0.003747  
2023-09-01                            -0.008615                  -0.000164  
2023-10-01                            -0.001206                   0.003235  
2023-11-01                             0.008482                   0.001633  
2023-12-01                            -0.010011                  -0.003992